pandas/core/arrays/arrow/array.py

```diff
@@ -392,6 +392,73 @@ def _from_sequence_of_strings(
             )
         return cls._from_sequence(scalars, dtype=pa_type, copy=copy)
 
+    def _cast_pointwise_result(self, values) -> ArrayLike:
+        if len(values) == 0:
+            # Retain our dtype
+            return self[:0].copy()
+
+        try:
+            arr = pa.array(values, from_pandas=True)
```

Comment:
This would convert `np.nan` in `values` to `NA`. Is that something we always want from this method?

Reply:
In main `pd.Series([1, 2, 3], dtype="float64[pyarrow]").map(lambda x: np.nan)` gives back np.float64.  If you cast that back to pyarrow backend those become NAs, so yes.  #62040 will give users the option to control this consistently.